### DISCUSSION TITLE
CompiledBinding to `SolidColorBrush.Color` doesn't work if source is StaticResource

### DISCUSSION THREAD

**Discussion by sinatrocious** [AvaloniaUI/Avalonia#15163] [General]
I want to override brush `TreeDataGridHeaderBackgroundPointerOverBrush` (used in `TreeDataGrid`) to have my own colors.

I have my own brush defined in resources and in WPF I'd [bind to it](https://stackoverflow.com/q/5486985/1997232), but I am getting error:

> Unable to resolve property or method of name 'Color' on type 'System.Object'.

To reproduce the issue just add somewhere in resources following:

```xaml
    <SolidColorBrush x:Key="brush1"
                     Color="#123456"/>
    
    <SolidColorBrush x:Key="brush2"
                     x:DataType="SolidColorBrush"
                     Color="{CompiledBinding Color, Source={StaticResource brush1}}"/>
```

Why error say `System.Object`? Do I miss something or is it intended? I found #5036, but it's closed already, so I guess it suppose to work, right?

I was thinking to simply define colors, but this didn't worked either: #15162

**Comment by timunie** [comment]
You may need to cast for CompiledBinding. Compiled bindings need to know the object type in advance.

**Reply by timunie** [reply]
https://docs.avaloniaui.net/docs/basics/data/data-binding/compiled-bindings#type-casting

**Reply by sinatrocious** [reply]
You are right, I don't need `x:DataType` but a cast:

```xaml
    <SolidColorBrush x:Key="brush2"
                     Color="{CompiledBinding (SolidColorBrush).Color, Source={StaticResource brush1}}"/>
```
